subsys/bluetooth/host/scan.c

```diff
@@ -716,22 +717,29 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 			      BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
 	     ((pending_per_adv_sync->sid != evt->sid) ||
 	      bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
-		struct bt_le_per_adv_sync_term_info term_info;
-
 		BT_ERR("Unexpected per adv sync established event");
+		/* Request terminate of pending periodic advertising in controller */
 		per_adv_sync_terminate(sys_le16_to_cpu(evt->handle));
 
+		unexpected_evt = true;
+	} else {
+		unexpected_evt = false;
+	}
+
+	if ((!unexpected_evt && evt->status != BT_HCI_ERR_SUCCESS) || unexpected_evt) {
```

Comment:
Btw, couldn't this be simplified to `if (unexpected_evt || evt->status != BT_HCI_ERR_SUCCESS)`?

Reply:
@jhedberg good suggestion, thanks!